### PR TITLE
chore: adds FDv2 source using composite datasource

### PR DESCRIPTION
**Pull request by tanderson-ld**
**Requirements**

- [x] I have added test coverage for new or changed functionality

- [x] I have followed the repository's [pull request submission guidelines](../blob/main/CONTRIBUTING.md#submitting-pull-requests)

- [ ] I have validated my changes against all supported platform versions
Not trivially possible yet, still need to hook up more code to what @kinyoklion has written.

---

> [!NOTE]
> Introduces FDv2 data source built from composable sources; refactors CompositeSource to delegate-based factories with async action processing and exhaustion/error reporting; adds extensive tests and test helpers.
> 
> - **FDv2 Data Source**:
>   - Add `FDv2DataSource` with initializer/synchronizer composition using `CompositeSource`.
>   - Implement action appliers: fast fallback for initializers, timed fallback/recovery for synchronizers, and blacklist-on-success/off.
> - **CompositeSource Refactor**:
>   - Replace `ISourceFactory`/`IActionApplierFactory` with delegate types `SourceFactory` and `ActionApplierFactory`.
>   - Process enqueued actions on a background thread; allow disposal while actions are queued.
>   - Report `Off` directly with optional `ErrorInfo`; emit exhaustion error when no sources remain.
>   - Build sinks via `FanOutDataSourceUpdates` and disableable tracker; maintain current entry for blacklist/remove.
> - **APIs**:
>   - Add delegates `SourceFactory` and `ActionApplierFactory`; remove corresponding interfaces.
> - **Tests**:
>   - Expand `CompositeSourceTest` for fallback, blacklist, disposal semantics, exhaustion, and disabled-source isolation.
>   - Add comprehensive `FDv2DataSourceTest` covering initializer/synchronizer flows, failures, and edge cases.
>   - Enhance `SourcesListTest` with non-circular end-of-list behavior.
> - **Test Utilities**:
>   - Extend `CapturingDataSourceUpdates` to record and wait for ordered status updates.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 370e94e774a91e79570dbcf19411afc0549e9bea. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: pkgs/sdk/server/src/Internal/FDv2DataSources/FDv2DataSource.cs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+{
+    internal static class FDv2DataSource
+    {
+        /// <summary>
+        /// Creates a new FDv2 data source.
+        /// </summary>
+        /// <param name="updatesSink">the sink that receives updates from the active source</param>
+        /// <param name="initializers">List of data source factories used for initialization</param>
+        /// <param name="synchronizers">List of data source factories used for synchronization</param>
+        /// <param name="fdv1Synchronizers">List of data source factories used for FDv1 synchronization</param>
```
Comment (kinyoklion) [inline pkgs/sdk/server/src/Internal/FDv2DataSources/FDv2DataSource.cs]: In configuration I have been doing this as a single item, but do we want to support a list?
